common/paramsbuilder/params.go

```diff
@@ -175,3 +177,34 @@ func (a APIModule) String() string {
 
 	return fmt.Sprintf("%s/%s", a.Label, a.Version)
 }
+
+// Metadata params sets metadata describing authentication.
+type Metadata struct {
+	Map          map[string]string
```

Comment:
can you add comments above `Map` and above `requiredKeys` to explain them? it's somewhat obvious from the name but good to be clear for future readers.